icgcget/clients/icgc/icgc_api.py

```diff
@@ -62,30 +62,27 @@ def get_manifest_id(manifest_id, api_url, repos=None):
 
 
 def get_manifest(file_ids, api_url, repos=None):
-
+    fields = '&fields=id,size,content,repoFileId'
     if repos:
-        request = api_url + 'manifests?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
-                            '"]}}}&repos=' + ','.join(repos) + '&unique=true&fields=id,size,content,repoFileId'
+        request = api_url + 'manifests' + filters(file_ids) + '&repos=' + ','.join(repos) + '&unique=true&' + fields
     else:
-        request = api_url + 'manifests?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
-                            '"]}}}&fields=id,size,content,repoFileId'
+        request = api_url + 'manifests' + filters(file_ids) + fields
     entity_set = call_api(request, api_url)
     return entity_set
 
 
 def get_metadata_bulk(file_ids, api_url):
-
     entity_set = []
-    request = api_url + 'repository/files?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
-                        '"]}}}&&from=1&size=10&sort=id&order=desc'
-    resp = call_api(request, api_url)
-    entity_set.extend(resp["hits"])
-    pages = resp["pagination"]["pages"]
-    page = 1
-    while page < pages:
-        request = api_url + 'files?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + \
-                  '"]}}}&&from={}&size=10&sort=id&order=desc'.format(page*10 + 1)
+    pages_available = True
+    while pages_available:
+        request = api_url + 'repository/files' + filters(file_ids) + '"&&from=1&size=10&sort=id&order=desc'
         resp = call_api(request, api_url)
-        entity_set.append(resp["hits"])
-        page = resp["pagination"]["pages"]
+        entity_set.extend(resp["hits"])
+        pages = resp["pagination"]["pages"]
+        page = resp["pagination"]["page"]
+        pages_available = page < pages
     return entity_set
+
+
+def filters(file_ids):
+    return '?filters={"file":{"id":{"is":["' + '","'.join(file_ids) + '"]}}}'
```

Comment:
But given that you are using `requests`, you should probably consider much simpler:

`http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls`